pkgs/expander/compile/side-effect.rkt

```diff
@@ -223,11 +307,11 @@
          fail-v)]
     [else fail-v]))
 
-(define (procedure-spec? e immutables-expr)
+(define (procedure-spec? e field-count)
```

Comment:
This is changed because specifying a field _makes_ the field immutable, it doesn't require it to be immutable.